cpp/src/copying/concatenate.cu

```diff
@@ -313,6 +313,12 @@ struct concatenate_dispatch {
   {
     return cudf::strings::detail::concatenate(views, mr, stream);
   }
+
+  template <typename T, std::enable_if_t<std::is_same<T, cudf::list_view>::value>* = nullptr>
+  std::unique_ptr<column> operator()()
```

Comment:
Just use explicit specialization instead of SFINAE here, it's much clearer and more concise. 
```suggestion
  template<>
  std::unique_ptr<column> operator()<list_view>()
```